test/ValidationFramework/Framework.validation.cpp

```diff
@@ -0,0 +1,291 @@
+using ostk::astro::trajectory::Segment;
+using ostk::astro::trajectory::Propagator;
+using ostk::astro::trajectory::Orbit;
+using ostk::astro::trajectory::LocalOrbitalFrameFactory;
+using ostk::astro::trajectory::LocalOrbitalFrameDirection;
+using ostk::astro::trajectory::state::NumericalSolver;
+using ostk::astro::trajectory::state::CoordinatesSubset;
+using ostk::astro::trajectory::state::CoordinatesBroker;
+using ostk::astro::trajectory::state::coordinatessubsets::CartesianPosition;
+using ostk::astro::trajectory::state::coordinatessubsets::CartesianVelocity;
+using ostk::astro::trajectory::state::NumericalSolver;
+using ostk::astro::trajectory::orbit::models::kepler::COE;
+using ostk::astro::Dynamics;
+using ostk::astro::flight::system::PropulsionSystem;
+using ostk::astro::flight::system::SatelliteSystem;
+using ostk::astro::dynamics::PositionDerivative;
+using ostk::astro::dynamics::CentralBodyGravity;
+using ostk::astro::dynamics::ThirdBodyGravity;
+using ostk::astro::dynamics::AtmosphericDrag;
+using ostk::astro::dynamics::Thruster;
+using ostk::astro::guidancelaw::ConstantThrust;
+using ostk::astro::guidancelaw::QLaw;
+using ostk::astro::solvers::FiniteDifferenceSolver;
+
+using ostk::astro::validation::Quantity;
+using ostk::astro::validation::Tool;
+using ostk::astro::validation::ToolComparison;
+using ostk::astro::validation::QuantityComparison;
+using ostk::astro::validation::MissionSequence;
+using ostk::astro::validation::CrossValidator;
+using ostk::astro::validation::Parser;
+
+class OpenSpaceToolkit_Astrodynamics_ValidationFramework
+    : public ::testing::TestWithParam<std::tuple<String, Array<ToolComparison>>>
+{
+   protected:
+    // void SetUp() override {}
+    const String pathToData = {"/app/test/ValidationFramework/data"};
+};
+
+TEST_P(OpenSpaceToolkit_Astrodynamics_ValidationFramework, ValidationTestRunner)
+{
+    // Access the test parameters
+    const auto parameters = GetParam();
+
+    const String scenarioName = std::get<0>(parameters);
+    const Array<ToolComparison> toolComparisons = std::get<1>(parameters);
+
+    std::cout << "\n\n\n\n\n\n\n\n\n\n----- Running test: " << scenarioName << "------" << std::endl;
+
+    // Instatiate mission sequence app to set up the scenario
+    MissionSequence missionSequence = {Parser::ParseYaml(pathToData, scenarioName)};
+
+    // Solve the scenario
+    missionSequence.run();
+
+    // Compare with each reference tool
+    for (const ToolComparison& toolComparison : toolComparisons)
+    {
+        const Tool tool = toolComparison.tool;
+        const Array<QuantityComparison> quantityComparisons = toolComparison.quantityComparisons;
+
+        // Compare output quantities with reference tool
+        const Array<VectorXd> errors =
+            missionSequence.compareResults(Parser::ParseCSV(pathToData, scenarioName, tool), toolComparison);
+
+        for (Size coordinateSubsetIndex = 0; coordinateSubsetIndex < quantityComparisons.getSize();
```

Comment:
Just a small detail:

Here your iterating first through `QuantityComparison` (outer loop), then through `States` (inner loop). E.g.:
```
- Test position[0]
- Test position[1]
- ...
- Test position[N]
- Test velocity[0]
- Test velocity[1]
- ...
```

Would it be more intuitive to test by state then by coordinate?
```
- Test position[0], test velocity[0], test mass[0]  // Tested whole State [0]
- Test position[1], test velocity[1], test mass[1]  // Tested whole State [1]
- ...
- 
- Test position[N], test velocity[N], test mass[N]  // Tested whole State [N]
```

It's no different than what you've done. It would just be swapping the loops. Just thought rows then cols might feel more _natural._

Reply:
Yeah I thought about that, but I chose the current layout to follow the structure of the `parameters` variable that holds all the test cases. The only way to have the test cases listed at the bottom of the file in a visually easy to understand way is to have the output quantity and tolerance in the same structure `QuantityComparison` (otherwise the quantity and the tolerance do not appear next to each other in each test case and it because harder to track which quantity should be compared with which tolerance). The looping structure I have reflects this structure by looping through all the states corresponding to one `QuantityComparison`.